### PR TITLE
Fix JS pickle implementation and BININT2 unsignedness

### PR DESCRIPTION
**Pull request by ehdr**
From a user reported bug. Python's BININT2 is unsigned 16-bit (0-65535), but our old implementation treats it as signed (-32768 to 32767), meaning integers 32768-65535 were decoded as negative numbers.

---

> [!NOTE]
> Corrects `BININT2` to use unsigned 16-bit encoding/decoding and adds boundary tests for 16-bit/32-bit integers.
> 
> - **Pickle codec (`modal-js/src/pickle.ts`)**:
>   - **Encoding**: Use `BININT2` for unsigned 16-bit (`0..0xffff`) instead of signed range.
>   - **Decoding**: Parse `BININT2` as unsigned (`uint16`) without sign extension; keep `BININT1`/`BININT4` behavior unchanged.
> - **Tests (`modal-js/src/pickle.test.ts`)**:
>   - Add integer boundary cases covering `BININT2` unsigned limits (e.g., `32768`, `65535`) and corresponding negative values via `BININT4`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d1ef4a4997ec7c860567424831a159b974470f36. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: modal-js/src/pickle.test.ts [hunk 1]
```diff
@@ -49,6 +49,32 @@ const testCases = [
...
+  {
+    name: "BININT2_65535 (max unsigned 16-bit)",
+    b64: "gASVBAAAAAAAAABN//8u",
+    expected: 65535,
+  },
+  {
+    name: "BININT2_-65535",
+    b64: "gASVBAAAAAAAAABN//8u",
+    expected: -65535,
+  },
```
[the remark this answers is not in the document]
Comment (ehdr) [reply modal-js/src/pickle.test.ts:77]: There is no BININT2_-65535 test case any more.

**Review by freider — APPROVED** [review]
lgtm. A bit scary that the old implementation was so explicitly doing it wrong
